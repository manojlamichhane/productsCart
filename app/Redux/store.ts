import { configureStore } from "@reduxjs/toolkit";
import productsReducer from "./features/Products/productsSlice";
import cartReducer from "./features/Products/cartSlice";
import authReducer from "./features/auth/userSlice";
import orderReducer from "./features/Products/orderSlice";

export const store = configureStore({
  reducer: {
    // states are accessed in useSelector using this name
    products: productsReducer,
    Cart: cartReducer,
    Auth: authReducer,
    Order: orderReducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
