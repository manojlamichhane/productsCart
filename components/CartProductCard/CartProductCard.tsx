import Image from "next/image";
import React from "react";
import { IconButton } from "..";
import { useAppDispatch } from "../../app/Redux/hook";
import {
  increaseQuantity,
  decreaseQuantity,
  deleteProduct,
} from "../../app/Redux/features/Products/cartSlice";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

const CartProductCard = ({ products }: { products: any[] }) => {
  const dispatch = useAppDispatch();

  const incrementQuantity = (id: string) => {
    dispatch(increaseQuantity(id));
    toast("Quantity Increased!", {
      position: "bottom-left",
      type: "success",
      theme: "colored",
    });
  };
  const decrementQuantity = (id: string) => {
    dispatch(decreaseQuantity(id));
    toast("Quantity Decreased!", {
      position: "bottom-left",
      type: "success",
      theme: "colored",
    });
  };

  const handleDelete = (id: string) => {
    dispatch(deleteProduct(id));
    toast("Item deleted from cart!", {
      position: "bottom-left",
      type: "success",
      theme: "colored",
    });
  };

  return (
    <div className="w-full">
      {products &&
        products.map((item) => {
          return (
            <div key={item._id} className="car-card m-2 ">
              <div className="flex flex-col md:flex-row justify-between items-center object-contain ">
                <Image
                  src={item.thumbnail}
                  alt="thumbnail"
                  width={50}
                  height={50}
                  className="object-contain m-5"
                />
                <div>
                  <h2 className="card__content-title text-xl font-bold">
                    {item.title}
                  </h2>
                  <p className="card__content-subtitle text-lg ">
                    {item.description}
                  </p>
                  <div className="flex flex-row items-center justify-between">
                    <div className="flex flex-row">
                      <p className="text-lg font-bold">$</p>
                      <p className=" text-lg font-bold">{item.price}</p>
                    </div>
                    <div className="flex flex-row justify-between items-center">
                      <p className=" text-lg font-bold">Quantity:</p>
                      <IconButton
                        source="/minusminus.png"
                        handleClick={() => decrementQuantity(item._id)}
                      />
                      <p className=" text-lg font-bold">{item.quantity}</p>
                      <IconButton
                        source="/plusplus.png"
                        handleClick={() => incrementQuantity(item._id)}
                      />
                    </div>
                    <IconButton
                      source="/Delete.png"
                      handleClick={() => handleDelete(item._id)}
                      styles="bg-white"
                      width={20}
                      height={20}
                    />
                  </div>
                </div>
              </div>
            </div>
          );
        })}
      <ToastContainer />
    </div>
  );
};

export default CartProductCard;
