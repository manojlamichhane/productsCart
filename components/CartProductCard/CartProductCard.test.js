import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import CartProductCard from "./CartProductCard";
import { Provider } from "react-redux";
import { store } from "./../../app/Redux/store";

describe("CartProductCard test", () => {
  test("Render check", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CartProductCard />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
