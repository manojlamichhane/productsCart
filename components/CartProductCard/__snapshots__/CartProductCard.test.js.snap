// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartProductCard test Render check 1`] = `
<DocumentFragment>
  <div
    class="w-full"
  >
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
